### ISSUE TITLE
Add Submodule `CqlToElmTests` into CQL repo

### ISSUE THREAD

**Issue by baseTwo** [FirelyTeam/firely-cql-sdk#449]
Before starting with this, pick up #428 first

CqlToElm test project are already in another repo https://github.com/cqframework/cql-tests, but a copy of the code exists in the cql repo. Remove the copy and include them as submodule in the cql repo. See related #440 

See my notes on submodules: https://firely.atlassian.net/wiki/x/EAD5Yg

**Comment by baseTwo** [comment]
Checked against `develop` @ `3810aa572`. There is real work left here, but the remedy as written — "remove the copy and include them as submodule" — is no longer the right move, so I am superseding rather than keeping.

**What carried over:**

- The duplication this issue is about still exists. The upstream corpus is vendored at `Cql/CqlToElmTests/Input/DQIC/` (15 XML files with upstream's exact filenames), plus a copy of upstream's schema at `Cql/CqlToElmTests/testSchema.xsd` and its generated deserializer `Cql/CqlToElmTests/testSchema.cs`. It is consumed by `XmlTest.GetTests()` at `Cql/CqlToElmTests/(tests)/XmlTest.cs:142-161`.
- Two concrete pieces of dead wiring are still in the tree:
  - `.gitmodules:1-4` still carries the commented-out `Cql/CqlToElmTests/cql-tests` stanza ("This submodule is not active, so commented out for now").
  - `Cql/CqlToElmTests/CqlToElmTests.csproj:44` still globs `cql-tests\tests\cql\*.xml`, a path that does not exist in the working tree.

**What did not carry over:**

- **The submodule was already tried and reverted.** It was added on 2024-05-20 (`043f89157` "Submodule the cql-tests repo"), re-pathed on 2024-08-05 (`b8cd37abd`), and commented out the same day (`fc745ca32`) — i.e. before this issue was even filed on 2024-08-06. The issue does not know that.
- **The local copy is no longer a copy.** It has diverged deliberately, in both directions:
  - Local-only file: `CqlAgeTest.xml`, added by `492a41026` (2024-08-21, "Added age calculation tests to XmlTests"). Upstream has no such file.
  - Upstream-only files we do not consume: `CqlAggregateTest.xml` and `CqlQueryTests.xml`.
  - 21 commits have modified `Cql/CqlToElmTests/Input/DQIC/`, 15 of them since 2025-01-01, and many are intentional spec corrections to upstream's expectations — e.g. `9a84f253c` "Fix IncludedInNullLeft XML assertion: null included in {2} → false per In spec", `e7c576b4d` "fix Round negative midpoint expectations to match CQL spec", `e5ef2c4d4` "Fix Split to use full string separator semantics".

  Replacing this with a submodule pinned to `cqframework/cql-tests` would discard all of that.
- **The prerequisite and related issues are closed.** "pick up #428 first" — #428 (Run integration tests in build pipeline) is closed; the related #440 (Better integration between CQL SDK and Integration Tests) is closed.
- **Upstream moved.** `cqframework/cql-tests` default branch is now `main`, not `master`, so any pin/URL guidance in the linked notes is stale.
- The design notes link (`https://firely.atlassian.net/wiki/x/EAD5Yg`) is Confluence-internal and cannot be relied on as the issue's specification.

The surviving, actionable concern is therefore: we maintain a silently-forked snapshot of an actively-developed upstream test corpus, with no recorded provenance, no divergence log, and dead submodule scaffolding left behind. That is what the replacement issue asks for.

Superseded by #1533.

Confidence: medium — the divergence, the dead `.gitmodules` stanza and the dead csproj glob are all verified in the tree and in git history; the call that "submodule" is the wrong remedy is a judgement based on two years of deliberate local edits, so push back if the intent was always to fork upstream and submodule the fork.
